### PR TITLE
Add users

### PR DESCRIPTION
**Pull request by nahiluhmot**
@tlunter @djosborne 

Add a users model, service, and controller to this app.

### CODE REVIEW

File: app/auth/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      self.table_name = 'users'
+
+      validates :name, presence: true
+      validates :email, presence: true, uniqueness: true
+
+      validates :password_salt, presence: true
+      validates :password_hash, presence: true
+
+      def as_json(opts = {})
+        super(opts.merge(except: %w(password_hash created_at updated_at)))
```
Comment (tlunter) [inline app/auth/models/user.rb]: The salt should probably be passed to except as well.

File: app/auth/services/users.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      end
+
+      def update(id, hash)
+        user = get_user(id)
+        email = hash['email']
+
+        if email.present? && (email != user.email) && exists?(email)
+          fail Errors::ConflictingModelOptions, "A user with email '#{email}' already exists"
+        end
+
```
Comment (tlunter) [inline app/auth/services/users.rb]: What if a password is being updated? The password being passed in will need to be hashed.

File: app/auth/models/user.rb (hunk 1 above)
Comment (nahiluhmot) [inline app/auth/models/user.rb]: I've seen arguments for both, but I suppose we don't have a use case for using the salt on the client. Will remove.

File: app/auth/services/users.rb (hunk 2 above)
Comment (nahiluhmot) [inline app/auth/services/users.rb]: Good call.

**Comment by nahiluhmot** [comment]
@tlunter 

* Rehash passwords on updates
* The users controller now prevents the `password_hash` and `password_salt` from being updated directly.

**Comment by tlunter** [comment]
:+1:

**Pull request merged**